network/src/validator_network/network_builder.rs

```diff
@@ -401,58 +396,41 @@ impl NetworkBuilder {
     pub fn build(mut self) -> NetworkAddress {
         use libra_network_address::Protocol::*;
 
-        let peer_id = self.peer_id;
         let protos = self.supported_protocols();
 
-        let trusted_peers = self.trusted_peers.clone();
         let authentication_mode = self
             .authentication_mode
             .take()
             .expect("Authentication Mode not set");
 
-        // formatting the listen address here is a temporary hack until `transport.rs`
-        // gets refactored and returns the fully rendered listen_addr in
-        // `Transport::listen_on()`.
-
-        // would like to do this in `build_with_transport` but ownership is very
-        // tricky here since the private keys aren't cloneable and we don't have
-        // a safe key container yet...
-        let unbound_listen_addr = self.listen_address.clone();
-        let unbound_len = unbound_listen_addr.len();
-        let unbound_listen_addr =
-            append_libranet_protocols(unbound_listen_addr, &authentication_mode);
+        let (key, maybe_trusted_peers) = match authentication_mode {
+            AuthenticationMode::ServerOnly(key) => (key, None),
```

Comment:
I'm confused, we are showing server only for the server side

probably starting to sound repetitive here, let's chat tomorrow :)